### ISSUE TITLE
Print on form not implemented?

### ISSUE THREAD

**Issue by KarlSevenSeven** [twinbasic/twinbasic#1025]
**Describe the bug**
Print on a form is not implemented.

**To Reproduce**
1. New EXE project
2. 
```
Private Sub Form_Load() Handles Form.Load
       Print "Loaded"
     End Sub

```
3. Runtime error

**Expected behavior**
Print working

**Desktop
 - Windows 10
 - twinBASIC 82

**Comment by MrBhop** [comment]
Unless i missed something, the `Printer` Class is also missing the `Print` Method.

**Comment by fafalone** [comment]
There's no `Printer.Print` in VB6.

**Comment by MrBhop** [comment]
Sorry, you're right. I never realized, because the app i'm maintaining uses this quite extensively so i just assumed this was standard stuff. 
I created a new Standard EXE project in vb6 to test and using `Printer.Print` works just fine. Would this count as backwards compatible behaviour, tB should support in your opinion?

To reproduce:
Paste the below code on a form with a button `Command1` and replace the `targetPrinter` with your desired Printer.
```vb6
Option Explicit

Private Sub Command1_Click()
    Const targetPrinter As String = "Microsoft Print to PDF"
    
    Dim i As Integer
    For i = 0 To Printers.Count - 1
        If Printers(i).DeviceName = targetPrinter Then
            Set Printer = Printers(i)
            GoTo PrinterSelected
        End If
    Next i
    MsgBox "Target Printer not available"
    Exit Sub
PrinterSelected:
    On Error Resume Next
    Printer.Print "Hello, World"
    Printer.EndDoc
End Sub

```

**Comment by fafalone** [comment]
Sorry, looks like it has a weird binding that just doesn't come up in the object browser or autocorrect or syntax highlight correctly (i.e. everything you check). 

But then that's the same in tB. I ran your test code in twinBASIC and it also worked fine there. Are you very far behind the latest version? This would have been implemented with general Printer object support a year or two ago.

**Comment by MrBhop** [comment]
Sorry, turns out my previous snippet was bit of a lie. I overlooked, that the code i was trying to port, used the `Call` keyword and that seems to be causing the issues.
Apparently if you use the `Call` keyword, tB throws a compiler error.
```vb6
' Works
Printer.Print "Hello, World!"

' Doesn't work - TB5027: Unrecognized member 'Print' on type '_Printer' [non-extensible object]
Call Printer.Print("Hello, World!")
```

I tried it with the latest version btw (870).

**Comment by fafalone** [comment]
You're able to use that in VB6? For me it's automatically changing it to `Printer.Print ("Hello, World!")` the second I click off the line. Could you post a small minimal reproduction VB6 project?

**Comment by MrBhop** [comment]
Yes, you're right. I keep forgetting that vb6 loves changing the code in subtle ways. You're 100% right that the IDE removes the `Call` keyword. In that case it makes perfect sense, that tB doesn't compile the code. I am very sorry for wasting your time with this.

**Comment by fafalone** [comment]
No trouble at all, it's always good to double check these things.

**Comment by wqweto** [comment]
Where does this Call Printer.Print comes from? IMO it wouldn’t be unreasonable for TB to handle it without hickups.

**Comment by MrBhop** [comment]
The `Call` was there because i missed, that vb6 removed it. I almost never look at/save the code in the vb6 IDE because the IDE is frankly a bit terrible and saving messes up the formatting and makes a billion changes because of casing and whatnot. So i totally missed it.

If you mean that tB could fix the code, like vb6 does that i would be against that. IMO those silent changes, vb6 does just lead to confusion. If you mean making `Call Printer.Print(someText)` valid syntax in tB i would support that. The current behaviour seems quite inconsistent.

**Comment by fafalone** [comment]
Should really support both `Call Printer.Print(s)` and just `Printer.Print(s)`. (The latter it automatically inserts a space, when it doesn't for other uses like that).

**Comment by MrBhop** [comment]
As it turns out `Form.Print` also already works (`beta 870`), just not in `Form.Activate` or any prior events. I guess something isn't being initialized early enough. The include projects show, that in vb6 you can call `Form.Print` in `Form.Activate` without issues but in tb it just doesn't do anything. In `Form.Resize` tb works as expected though. PicureBoxes behave the same in tb, btw. I didn't test those in vb6 but i'd assume that they would behave like Forms.

[tbFormPrinting.zip](https://github.com/user-attachments/files/21712295/tbFormPrinting.zip)
[vb6FormPrinting.zip](https://github.com/user-attachments/files/21712296/vb6FormPrinting.zip)

**Comment by fafalone** [comment]
Yup definitely a bug with Form_Activate, that should be posted as a new bug report though, don't want it to get buried in this ancient thread that should probably be closed since the method exists now.

Though it doesn't work in Form_Load in VB6 like the original post in this issue suggests (or Initialize), so it's just Activate.

**Comment by fafalone** [comment]
Closing this since it's generally working and we have a separate issue for the bug.
